lib/persistence/object.rb

```diff
@@ -670,6 +670,17 @@ def generate_inserts_for_two_phase(object, use_customer_email_param)
           save_pending_file(customer['name'], 'customers', customer)
         end
 
+      elsif payload_key.pluralize == 'creditmemos'
+
+        if !use_customer_email_param
+          object_dup = object.dup
+          object_dup['list_id'] = nil
```

Comment:
Why are we removing the list_id here?

Reply:
Also, should we allow us to set up the customer object in a FL transform: like
```if use_customer_object?
            customer = object['customer']
          else```
?

Reply:
It was trying to use the credit memo's list_id to build the customer and I wasn't sure if the object was being used somewhere else. To be safe, I duplicated the hash and set the list_id to nil

Reply:
And yeah, we can add that! I'll test it out it QB Dev to make sure its able to set it properly

Reply:
Oh I see - `object` is the creditmemo... I thought it was the customer. Yeah that makes sense!

Reply:
Now that I think about it, this might be affecting the other invocations of `build_customer_from_order` when they go through mods? It seems harmless though since the customer is in the object and could error out without affecting the mod

Reply:
Not sure I follow what you mean here...

Reply:
All the calls I saw to `build_customer_from_order` passes in the whole object so if there's something like an invoice with a list id already, the method would fail since it'll try to use that list id when building the customer